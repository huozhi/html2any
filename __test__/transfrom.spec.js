const html1 = require('./mock/html1')
const tokenizer = require('../src/tokenizer')
const parser = require('../src/parser')
const transform = require('../src/transform')
const React = require('react')
const ReactDOMServer = require('react-dom/server')

function rule (node, children) {
  if (typeof node === 'string') {
    return node.replace(/&amp;/, '&')
  }
  const {name, attribues} = node
  if (attribues.hasOwnProperty('class')) {
    attribues.className = attribues.class
    delete attribues.class
  }
  let elem
  switch (name) {
    case 'p': {
      elem = {
        tag: 'h1',
        props: attribues
      }
      break
    }
    case 'b': {
      elem = {
        tag: 'h2',
        props: attribues
      }
      break
    }
    case 'div': {
      elem = {
        tag: 'div',
        props: attribues
      }
      break
    }
  }

  if (!elem || !elem.tag) {
    return null
  }

  // add react index prop
  if (typeof node.index === 'number') {
    Object.assign(elem.props, {key: node.index})
  }
  return React.createElement(elem.tag, elem.props, children)
}

const ast = parser(tokenizer(html1))[0]
const output = transform(ast, rule)

console.log(ReactDOMServer.renderToStaticMarkup(output))
