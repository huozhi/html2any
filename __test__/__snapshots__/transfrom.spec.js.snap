// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform works well on html1 with customized rule 1`] = `"<div><hr/><img src=\\"a.jpg\\" data-index=\\"2\\"/><h1 id=\\"x34\\">1234 &amp;amp;</h1><div id=\\"container\\" class=\\"d3 d2\\"><input value=\\"1 2 3\\" data-id=\\"xx\\" readonly=\\"\\"/></div></div>"`;

exports[`transform works well on html2 with customized rule 1`] = `"<div><h1><br/></h1><h1 data-mce-style=\\"text-align: center; font-size: 15px;\\"><img src=\\"https://www.tinymce.com/images/glyph-tinymce@2x.png\\" alt=\\"Tiny M C E\\" width=\\"110\\" height=\\"97\\" data-mce-src=\\"images/glyph-tinymce@2x.png\\"/></h1><h1>This demo includes these TinyMCE Premium<span data-mce-bogus=\\"1\\" data-mce-word=\\"Plugins\\" data-mce-index=\\"0\\" class=\\"mce-spellchecker-word\\">Plugins</span>:<a href=\\"https://www.tinymce.com/powerpaste\\" data-mce-href=\\"powerpaste\\" data-mce-linkchecker-status=\\"valid\\">PowerPaste</a>(improved Word paste),<a href=\\"https://www.tinymce.com/accessibility-checker\\" data-mce-href=\\"accessibility-checker\\" data-mce-linkchecker-status=\\"valid\\">Accessibility Checker</a>,<a href=\\"https://www.tinymce.com/spell-checker-pro\\" data-mce-href=\\"spell-checker-pro\\" data-mce-linkchecker-status=\\"valid\\">Spell Checker Pro</a>&amp;amp; file management with<a href=\\"http://www.moxiemanager.com\\" data-mce-href=\\"http://www.moxiemanager.com\\" data-mce-linkchecker-status=\\"valid\\">MoxieManager</a>.</h1><h1 data-mce-style=\\"text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;\\">TinyMCE is a platform indepen<strong>dent web-</strong>based JavaScript HTML WYSIWYG<br/>editor control released as open source<em>under</em>LGPL.</h1><h1 data-mce-style=\\"text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;\\"><em>TinyMCE enables you to convert HTML<span data-mce-bogus=\\"1\\" data-mce-word=\\"textarea\\" data-mce-index=\\"0\\" class=\\"mce-spellchecker-word\\">textarea</span>fields or other HTML elements to editor instances.</em></h1><h1><br/></h1></div>"`;
